trin-beacon/src/storage.rs

```diff
@@ -223,8 +223,16 @@ impl ContentStore for BeaconStorage {
         }
     }
 
-    fn put<V: AsRef<[u8]>>(&mut self, key: Self::Key, value: V) -> Result<(), ContentStoreError> {
-        self.store(&key, &value.as_ref().to_vec())
+    fn put<V: AsRef<[u8]>>(
+        &mut self,
+        key: Self::Key,
+        value: V,
+    ) -> Result<Vec<(Self::Key, Vec<u8>)>, ContentStoreError> {
+        match self.store(&key, &value.as_ref().to_vec()) {
```

Comment:
Same as state store, this could be:
`self.store(...).and(Ok(vec![]))`.

Or if you prefer this:
`self.store(...).map(|_| vec![])`.